chain/sync.go

```diff
@@ -466,16 +466,25 @@ func (syncer *Syncer) ValidateTipSet(ctx context.Context, fts *store.FullTipSet)
 		return nil
 	}
 
+	var futures []async.ErrorFuture
 	for _, b := range fts.Blocks {
```

Comment:
First time in a long time, this lint warning saves the day.

`b` here is a reference, that will get updated to new value every loop iteration, and it is not captured correctly by closure.
Closure captures the reference `b`, not a copy `b`.
This effect could lead to us not validating all blocks.

The solution is to rebind `b` to a scoped variable:
```
b := b
```

Reply:
Or pass it as an argument to the closure but that is not simple here.

Reply:
Example: https://play.golang.org/p/yfHLJU5zkFZ

Reply:
:+1: for the clear explanation, go's syntactic sugar is evil